### PR TITLE
795 spice instrument pointing at specified time

### LINKED ISSUE #795: SPICE - Instrument pointing at specified time
### Algorithm Description:
Write code query spice in order to get an instrument's look direction at a specified time. To do this, we need to understand each instrument's frame definition and how it relates to the boresight. That info may be in the frame kernel imap_wkcp.tf

GLOWS requests this in "Cartesian heliocentric coordinates in ecliptic reference system"

### PR DESCRIPTION
**Pull request by tmplummer**
# Change Summary

## Overview
Add preliminary function for computing instrument pointing using spice.

## Updated Files
- imap_processing/spice/geometry.py
   - Modify frame_transform function to allow multiple time inputs with a single position vector.
   - Add simple lookup for getting instrument boresight vectors.
   - Add instrument pointing function that can return Lat/Lon or Cartesian coordinates
- imap_processing/tests/spice/test_geometry.py
   - Parameterize test for frame_transform and add coverage for multiple time inputs with single position vector
   - Add test coverage for new `instrument_pointing` function

Closes: #795

### CODE REVIEW

File: imap_processing/spice/geometry.py [hunk 1]
```diff
@@ -320,3 +337,42 @@ def get_rotation_matrix(
...
+        The instrument pointing at the specified times.
+    """
+    pointing = frame_transform(
+        et, BORESIGHT_LOOKUP[instrument.name], instrument, to_frame
+    )
+    if cartesian:
+        return pointing
+    if isinstance(et, typing.Collection):
+        return np.array([spice.reclat(vec)[1:] for vec in pointing])
+    return np.array(spice.reclat(pointing)[1:])
```
Comment (tmplummer) [inline imap_processing/spice/geometry.py]: `spice.reclat()` returns radians. These should probably be converted to degrees.

File: imap_processing/spice/geometry.py [hunk 2]
```diff
@@ -62,6 +62,24 @@ class SpiceFrame(IntEnum):
     IMAP_GLOWS = -43750
 
 
+# TODO: Update boresight for in-situ instruments
+# TODO: Confirm ENA boresight vectors
+BORESIGHT_LOOKUP = {
```
Comment (tmplummer) [inline imap_processing/spice/geometry.py:67]: This is my low effort initial approach to defining instrument boresights. I am open to suggestions of how to make this better, where to move this definition, etc. I still don't know if we should eventually create an instrument kernel that defines these. I need to get a better understanding of all of the requirements from all of the instruments before making that decision.

File: imap_processing/spice/geometry.py (hunk 1 above)
Comment (greglucas) [inline imap_processing/spice/geometry.py]:
```suggestion
        return np.rad2deg([spice.reclat(vec)[1:] for vec in pointing])
    return np.rad2deg(spice.reclat(pointing)[1:])
```

File: imap_processing/spice/geometry.py (hunk 2 above)
Comment (greglucas) [reply imap_processing/spice/geometry.py:67]: Do we have to use `.name` or can it just be the enumeration itself as the lookup?

**Review by greglucas — APPROVED** [review]

**Review by tech3371 — APPROVED** [review]
LTGM with my limited knowledge.

**Pull request merged**